### PR TITLE
Sort by _lastUpdated in Sql server

### PR DESCRIPTION
**Pull request by rbans96**
## Description
Adds Support for sorting(ASC and DESC) by _lastUpdated in SqlServer

## Related issues
Addresses #68732

## Testing
Adds e2e tests

### CODE REVIEW

File: src/Microsoft.Health.Fhir.SqlServer/Features/Search/SortUtils.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        public static SortOrder GetSortOrderByParameterName(SearchOptions searchOptions, string parameter)
+        {
+            var sortOrder = SortOrder.Ascending;
+            if (searchOptions.Sort?.Count > 0)
+            {
+                sortOrder = searchOptions
+                    .Sort
+                    .Where(x => string.Equals(x.searchParameterInfo.Name, parameter, StringComparison.OrdinalIgnoreCase))
+                    .Select(s => s.sortOrder).First();
```
Comment (johnstairs) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Search/SortUtils.cs:22]: This will throw if there are no matches

Comment (johnstairs) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Search/SortUtils.cs:22]: What if they try to order by another parameter?

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/SearchTestsBase.cs [hunk 2]
```diff
@@ -27,33 +30,83 @@ protected SearchTestsBase(TFixture fixture)
...
+            ValidateBundle(bundle, selfLink, true, expectedResources);
 
             return bundle;
         }
 
+        protected async Task<Bundle> ExecuteAndValidateBundle(string searchUrl, string selfLink, bool sort, params Resource[] expectedResources)
+        {
+            FhirResponse<Bundle> firstBundle = await Client.SearchAsync(searchUrl);
+
+            var firstExpectedResources = expectedResources.Length > 10 ? expectedResources.ToList().GetRange(0, 10).ToArray() : expectedResources;
```
Comment (brendankowitz) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/SearchTestsBase.cs:54]: Can we have `10` as a variable that is also used to set the `_count` in search, just so this paging seems more obvious

Comment (brendankowitz) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/SearchTestsBase.cs:54]: Alternatively, maybe pull this out into a pageSize field

File: src/Microsoft.Health.Fhir.SqlServer/Features/Search/SortUtils.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+namespace Microsoft.Health.Fhir.SqlServer.Features.Search
+{
+    public static class SortUtils
+    {
+        private static readonly IList<string> SupportedParameters = new List<string>
+            {
+                KnownQueryParameterNames.LastUpdated,
+            };
+
+        public static SortOrder GetSortOrderForSupportedParam(SearchOptions searchOptions)
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Search/SortUtils.cs:19]: I think we're good on functionality in this PR, just a few more thoughts on this class.
 - Add some documentation for what this function is doing
 - Probably a good candidate for an extension method, also rename so its more clear what its doing `public static SortOrder GetFirstSortOrderForSupportedParam(this SearchOptions searchOptions)`

File: src/Microsoft.Health.Fhir.SqlServer/Features/Search/SortUtils.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+                else
+                {
+                    throw new SearchParameterNotSupportedException(string.Format(Core.Resources.SearchParameterNotSupported, sortOptions.searchParameterInfo.Name));
+                }
+            }
+
+            return sortOrder;
+        }
+
+        public static bool IsSearchParamterSupported(string parameter)
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Search/SortUtils.cs:38]: Could also be an extension on SearchParameterInfo: `public static bool IsSearchParamterSupported(this SearchParameterInfo parameter)`

File: src/Microsoft.Health.Fhir.SqlServer/Features/Search/SortUtils.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+        public static SortOrder GetSortOrderForSupportedParam(SearchOptions searchOptions)
+        {
+            var sortOrder = SortOrder.Ascending;
+
+            foreach (var sortOptions in searchOptions.Sort)
+            {
+                if (IsSearchParamterSupported(sortOptions.searchParameterInfo.Name))
+                {
+                    sortOrder = sortOptions.sortOrder;
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Search/SortUtils.cs:27]: If we indicate its only getting _first_ we can probably return here

File: src/Microsoft.Health.Fhir.SqlServer/Features/Search/SortUtils.cs (hunk 3 above)
Comment (rbans96) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Search/SortUtils.cs]: Good point. Fixed

**Pull request merged**